crates/forge-runner/src/cairo_runner/sierra_casm_runner.rs

```diff
@@ -81,80 +86,85 @@ impl SierraCasmRunner {
+
+        let (cells, ap) = casm_run::run_function(
             vm,
             instructions,
             builtins,
-            |context| {
-                let vm = context.vm;
-                // Create the builtin cost segment, with dummy values.
-                let builtin_cost_segment = vm.add_memory_segment();
-                for token_type in CostTokenType::iter_precost() {
-                    vm.insert_value(
-                        (builtin_cost_segment + (token_type.offset_in_builtin_costs() as usize))
-                            .unwrap(),
-                        Felt252::from(token_gas_cost(*token_type)),
-                    )
-                    .map_err(|e| Box::new(e.into()))?;
-                }
-                // Put a pointer to the builtin cost segment at the end of the program (after the
-                // additional `ret` statement).
-                vm.insert_value(
-                    (vm.get_pc() + context.data_len).unwrap(),
-                    builtin_cost_segment,
-                )
-                .map_err(|e| Box::new(e.into()))?;
-                Ok(())
-            },
-            runtime,
+            initialize_vm,
+            hint_processor,
             hints_dict,
         )?;
-        let mut results_data = self.get_results_data(func, &cells, ap);
-        // Handling implicits.
-        let mut gas_counter = None;
-        results_data.retain_mut(|(ty, values)| {
-            let info = self.get_info(ty);
-            let generic_ty = &info.long_id.generic_id;
-            if *generic_ty == GasBuiltinType::ID {
-                gas_counter = Some(values.remove(0));
-                assert!(values.is_empty());
-                false
-            } else {
-                *generic_ty != RangeCheckType::ID
-                    && *generic_ty != BitwiseType::ID
-                    && *generic_ty != EcOpType::ID
-                    && *generic_ty != PedersenType::ID
-                    && *generic_ty != PoseidonType::ID
-                    && *generic_ty != SystemType::ID
-                    && *generic_ty != SegmentArenaType::ID
-            }
-        });
+        let (results_data, gas_counter) = Self::get_results_data(&return_types, &cells, ap);
         assert!(results_data.len() <= 1);
+
         let value = if results_data.is_empty() {
             // No result type - no panic.
             RunResultValue::Success(vec![])
         } else {
-            let [(ty, values)] = <[_; 1]>::try_from(results_data).ok().unwrap();
-            self.handle_main_return_value(&ty, values, &cells)
+            let (ty, values) = results_data[0].clone();
+            if let Some(inner_ty) = self.maybe_inner_type(&ty, func) {
+                let inner_ty_size = self.type_sizes[&inner_ty];
+                Self::handle_main_return_value(Some(inner_ty_size), values, &cells)
+            } else {
+                Self::handle_main_return_value(None, values, &cells)
+            }
```

Comment:
```suggestion
            let inner_type_size = self.maybe_inner_type(&ty, func)
                .map(|x| x.type_sizes[&inner_ty]);
            Self::handle_main_return_value(inner_type_size, values, &cells)
```
This is a nitpick/kownledge sharing, no need to change.